packages/sw-cacheable-response/src/lib/cacheable-response.js

```diff
@@ -65,14 +65,19 @@ class CacheableResponse {
    * Checks a response to see whether it's cacheable or not, based on the
    * configuration of this object.
    *
-   * @private
    * @param {Object} input
    * @param {Response} input.response The response that might be cached.
+   * @param {Request} [input.request] Optionally, the request that led to the
+   *        response.
    * @return {boolean} `true` if the `Response` is cacheable, based on the
    *          configuration of this object, and `false` otherwise.
    */
-  isResponseCacheable({response} = {}) {
+  isResponseCacheable({request, response} = {}) {
     assert.isInstance({response}, Response);
+    if (request) {
+      assert.isInstance({request}, Request);
+    }
+
```

Comment:
Not a blocker: I wonder if it would be more clear for us to be rejecting a promise for isResponseCacheable style interfaces rather than completely throwing, but I guess we would still need to return from the assertion so perhaps this is fine after-all.